pydantic/main.py

```diff
@@ -804,3 +734,155 @@ def validate_model(  # noqa: C901 (ignore complexity)
+    skip_defaults: bool,
+    as_type: Optional[Type[Any]] = None,
+) -> Any:
+    if isinstance(v, BaseModel):
+        if to_dict:
+            return v.dict(
+                by_alias=by_alias, skip_defaults=skip_defaults, include=include, exclude=exclude, as_type=as_type
+            )
+        else:
+            return v.copy(include=include, exclude=exclude)
+    if not (include or exclude):
+        return _get_value_fast(v, to_dict, by_alias, skip_defaults, as_type)
+
+    value_exclude = ValueItems(v, exclude) if exclude else None
+    value_include = ValueItems(v, include) if include else None
+
+    if isinstance(v, dict):
+        return {
+            k_: _get_value(
+                v_,
+                to_dict=to_dict,
+                by_alias=by_alias,
+                skip_defaults=skip_defaults,
+                include=value_include and value_include.for_element(k_),
+                exclude=value_exclude and value_exclude.for_element(k_),
+                as_type=as_type,
+            )
+            for k_, v_ in v.items()
+            if (not value_exclude or not value_exclude.is_excluded(k_))
+            and (not value_include or value_include.is_included(k_))
+        }
+
+    elif isinstance(v, (list, set, tuple)):
+        return type(v)(
+            _get_value(
+                v_,
+                to_dict=to_dict,
+                by_alias=by_alias,
+                skip_defaults=skip_defaults,
+                include=value_include and value_include.for_element(i),
+                exclude=value_exclude and value_exclude.for_element(i),
+                as_type=as_type,
+            )
+            for i, v_ in enumerate(v)
+            if (not value_exclude or not value_exclude.is_excluded(i))
+            and (not value_include or value_include.is_included(i))
+        )
+
+    else:
+        if as_type is not None:
+            return parse_as_type(v, as_type)
+        return v
+
+
+@no_type_check
+def _get_value_fast(
+    v: Any, to_dict: bool, by_alias: bool, skip_defaults: bool, as_type: Optional[Type[Any]] = None
+) -> Any:
+    if isinstance(v, BaseModel):
+        if to_dict:
+            return v.dict(by_alias=by_alias, skip_defaults=skip_defaults, as_type=as_type)
+        else:
+            return v.copy()
+
+    if isinstance(v, dict):
+        return {
+            k_: _get_value_fast(v_, to_dict=to_dict, by_alias=by_alias, skip_defaults=skip_defaults, as_type=as_type)
+            for k_, v_ in v.items()
+        }
+
+    elif isinstance(v, (list, set, tuple)):
+        return type(v)(
+            _get_value_fast(v_, to_dict=to_dict, by_alias=by_alias, skip_defaults=skip_defaults, as_type=as_type)
+            for i, v_ in enumerate(v)
+        )
+
+    else:
+        if as_type is not None:
+            return parse_as_type(v, as_type)
+        return v
+
+
+def _requires_casting(obj: Any, old_field: ModelField, new_field: ModelField) -> bool:
```

Comment:
I basically need to include this function in this module since it requires `BaseModel` to be imported, but also needs to be used in this module.

If we move this to `tools.py`, it would need to include the `from pydantic.main import BaseModel` inside the function call, adding an unfortunate amount of overhead.